auv_navigation/auv_localization/scripts/pressure_odometry_node.py

```diff
@@ -42,19 +46,75 @@ def __init__(self):
         rospy.loginfo(
             f"{pressure_odometry_colored} : depth covariance: {self.depth_calibration_covariance}"
         )
+
+        self.imu_data = None
+        self.imu_subscriber = rospy.Subscriber(
+            "imu/data", Imu, self.imu_callback, tcp_nodelay=True
+        )
+
         self.depth_subscriber = rospy.Subscriber(
             "depth", Float32, self.depth_callback, tcp_nodelay=True
         )
 
+        self.base_to_pressure_translation = None
+        rate = rospy.Rate(1.0)
+
+        while not rospy.is_shutdown() and self.base_to_pressure_translation is None:
+            try:
+                self.base_to_pressure_translation, _ = self.transformer.get_transform(
+                    "taluy/base_link", "taluy/base_link/external_pressure_sensor_link"
+                )
+            except Exception as e:
+                rospy.logwarn(f"Waiting for transform: {e}")
+                rate.sleep()
+
+    def imu_callback(self, imu_msg):
+        self.imu_data = imu_msg
+
+    def get_base_to_pressure_height(self):
+        try:
+            if self.base_to_pressure_translation is None:
+                rospy.logerr_throttle(10, "Transform not available.")
+                return 0.0
+
+            if self.imu_data is None:
+                rospy.logwarn_throttle(
+                    10, "No IMU data received yet. Using default orientation."
+                )
+                return self.base_to_pressure_translation[0, 2]
+
+            else:
+                orientation = self.imu_data.orientation
+                quaternion = [
+                    orientation.x,
+                    orientation.y,
+                    orientation.z,
+                    orientation.w,
+                ]
+
+                # Convert quaternion to 3x3 rotation matrix
+                rotation_matrix = tf.transformations.quaternion_matrix(quaternion)[
+                    :3, :3
+                ]
+
+                sensor_offset = np.array(self.base_to_pressure_translation[0])
+
+                # Rotate the translation vector using the rotation matrix
+                rotated_vector = rotation_matrix.dot(sensor_offset)
+
+                return rotated_vector[2]
+
+        except Exception as e:
+            rospy.logerr_throttle(10, f"Error in get_base_to_pressure_height: {e}")
+            return 0.0
```

Comment:
The `return 0.0` will be unnecessary after removing the try except block since nothing is expected to raise an exception here. 

But in general these kind of default returns must be used with caution since in this function it may cause the EKF to jump between values, which is something we don't want. We must handle all the cases we can, if not just let the `assert` raise an exception, or you manually raise one, which will stop the execution.